codemod-com/Commons
# refactor: reorganize folder structure, add pnpm workspace, update dependencies, and edit .gitignore

#### 📚 Description
- Reorganized project folder structure for better maintainability
- Added pnpm workspace for efficient package management
- Moved all legacy code into a `codemods` folder
- Added default packages and necessary tools in the `tooling` directory
- Edited .gitignore to include new patterns, including files like node_modules and lock files

This PR refactors the project to improve maintainability by reorganizing the folder structure. All legacy code has been moved into a dedicated `codemods` folder for better organization. Additionally, I have introduced a `pnpm workspace` for more efficient package management, added necessary tools in the `tooling` directory, and updated the `.gitignore` file to ensure that files like `node_modules` and lock files are properly ignored.

#### 🧪 Test Plan
To verify these changes, follow the steps below:
1. Clone the repository.
2. Run `pnpm install` to set up the workspace.
3. Check that the `codemods` folder contains all the legacy code.
4. Ensure that the `.gitignore` is correctly ignoring `node_modules`, lock files, and any other unnecessary files.
5. Run any relevant tests to confirm the setup is correct.

#### 📄 Documentation to Update
No documentation updates are required for this change. However, if necessary, I'll update the README to reflect the new project structure.
